game-core/src/main/java/games/strategy/util/Util.java

```diff
@@ -71,4 +74,23 @@ public static String sha512(final String input) {
 
     return p.negate();
   }
+
+  public static String getFromSetting(final ClientSetting<char[]> setting) {
+    final char[] charArray = setting.getValueOrThrow();
+    try {
+      return new String(charArray);
+    } finally {
+      Arrays.fill(charArray, '\0');
+    }
+  }
+
+  public static boolean isInt(final String string) {
+    Preconditions.checkNotNull(string);
+    try {
+      Integer.parseInt(string);
+      return true;
+    } catch (final NumberFormatException e) {
```

Comment:
Control flow by exception is a smell.  Why not just use a regex?  Also, unit tests would be trivial to add for this method.

Reply:
I probably [wrote too much python code in the past](https://docs.python.org/3.4/glossary.html#term-eafp) ^^